src/cljdoc/render/api_searchset.clj

```diff
@@ -29,27 +27,119 @@
   [version-entity name-of-ns def-name]
   (str (path-for-namespace version-entity name-of-ns) "#" def-name))
 
-(defn- elements->doc-segments
-  "Does the tricky work of walking through the Jsoup node siblings and collecting
-  headers and their text. Important considerations:
+(defn- soup-elem?
+  "Returns true if Jsoup elem is an html element (as oppposed to a text node)"
+  [elem]
+  (instance? org.jsoup.nodes.Element elem))
 
-  1. Documents do not necessarily start with a header.
-  2. Headers do not necessarily have text between them and the next header or the
-     end of the document.
-  3. Flexmark makes sure that all markdown headers have anchors, meaning this only works for markdown and not asciidoc."
-  [elements doc-title]
+(defn heading-elem?
+  [elem]
+  (and
+   (soup-elem? elem)
+   (#{"h1" "h2" "h3" "h4" "h5" "h6"} (.tagName elem))
+   (= "a" (some-> elem .children first .tagName))))
+
+(defn- adoc-section? [elem]
+  (and (soup-elem? elem)
+       (= "div" (.tagName elem))
+       (.hasAttr elem "class")
+       (re-matches #"sect[1-9]" (.attr elem "class"))
```

Comment:
do we need anchors on this regex? currently it would match "bisect123"

Reply:
Thanks for the careful review!

I think `re-matches` must match the whole string?
If we were using `re-find` I think we'd need those anchors.

Reply:
Yeah, just double-checked in the REPL:

```Clojure
Clojure 1.11.1
user=> (re-matches #"sect[1-9]" "bisect123")
nil
user=> (re-matches #"sect[1-9]" "sect123")
nil
user=> (re-matches #"sect[1-9]" "sect9")
"sect9"
user=> (re-find #"sect[1-9]" "bisect123")
"sect1"
```